app/views/contexts/hub_context.py

```diff
@@ -136,12 +146,17 @@ def _get_rows(self, enabled_section_ids) -> List[Mapping[str, Union[str, List]]]
 
                 if repeating_list:
                     for list_item_id in self._list_store[repeating_list].items:
+
+                        list_item_index = len(rows)
+
                         rows.append(
                             self._get_row_for_repeating_section(
-                                section_id, list_item_id
+                                section_id, list_item_id, list_item_index
                             )
                         )
                 else:
-                    rows.append(self._get_row_for_section(section_title, section_id))
+                    rows.append(
+                        self._get_row_for_section(section_title, section_id, section_id)
+                    )
```

Comment:
I don't think we should do any this, I find this really confusing as we are passing section ID twice, which may differ if it is repeating etc.

I think we should keep all this as is, and pass in the list_item_id to `get_row_context_for_section` which can call a method to return an appropriate title id based of whether there is a list item id and whether it is repeating. It means, we have to call `self._schema.get_repeating_list_for_section(section_id)` an extra time but given these would be cached in future, the performance should be negligible.

Reply:
Yeah I get that passing section ID in twice looks a bit odd, it just allowed me to conveniently use it for two different things. While still allowing `get_row_context_for_section` to be used for sections and repeating sections.

Reply:
My comment above about passing `list_index` should resolve this.